provider/bitbucket_test.go

```diff
@@ -0,0 +1,183 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/karlderkaefer/cdk-notifier/config"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"sync"
+	"testing"
+)
+
+type MockBitbucketRepositoryService struct {
+	sync.RWMutex
+	comments []*BitbucketComment
+}
+
+func (m *MockBitbucketRepositoryService) ListComments(ctx context.Context, owner string, repo string, prId int64, opts *ListCommentOptions) (*BitbucketComments, *http.Response, error) {
+	comments := &BitbucketComments{}
+	for _, comment := range m.comments {
+		comments.Values = append(comments.Values, *comment)
+	}
+	return comments, nil, nil
+}
+
+func (m *MockBitbucketRepositoryService) EditComment(ctx context.Context, owner string, repo string, prId int64, commentID int64, comment *BitbucketComment) (*BitbucketComment, *http.Response, error) {
+	m.Lock()
+	defer m.Unlock()
+	for i, localComment := range m.comments {
+		if *localComment.Id == commentID {
+			m.comments[i] = &BitbucketComment{
+				Id:      &commentID,
+				Content: comment.Content,
+				Links:   comment.Links,
+			}
+			return m.comments[i], nil, nil
+		}
+	}
+	return nil, nil, fmt.Errorf("could not find comment with id %d in database %v", commentID, m.comments)
+}
+
+func (m *MockBitbucketRepositoryService) CreateComment(ctx context.Context, owner string, repo string, prId int64, comment *BitbucketComment) (*BitbucketComment, *http.Response, error) {
+	m.Lock()
+	defer m.Unlock()
+	id := int64(len(m.comments))
+	comment.Id = &id
+	m.comments = append(m.comments, comment)
+	return comment, nil, nil
+}
+
+func (m *MockBitbucketRepositoryService) DeleteComment(ctx context.Context, owner string, repo string, prId int64, commentID int64) (*BitbucketComment, *http.Response, error) {
+	var index int
+	var found bool
+	for i, comment := range m.comments {
+		if *comment.Id == commentID {
+			index = i
+			found = true
+		}
+	}
+	if !found {
+		return nil, nil, fmt.Errorf("could not find comment to delete with id %d", commentID)
+	}
+	copy(m.comments[index:], m.comments[index+1:])
+	m.comments[len(m.comments)-1] = nil
+	m.comments = m.comments[:len(m.comments)-1]
+	return nil, nil, nil
+}
+
+func defaultTestBitbucketProvider(comments []*BitbucketComment) *BitbucketProvider {
+	mock := &MockBitbucketRepositoryService{comments: comments}
+	return &BitbucketProvider{
+		Service:        mock,
+		Context:        context.Background(),
+		Config:         config.NotifierConfig{TagID: "test-tag", DeleteComment: true},
+		CommentContent: "test-content",
+	}
+}
+
+func TestBitbucketProvider_CreateAndDeleteComment(t *testing.T) {
```

Comment:
Meh... `TestCreateAndDeleteBitbucketComment`

Reply:
the default convention at least in intellij is `classname_methodname` it should be `TestBitbucketProvider_CreateComment` and `TestBitbucketProvider_DeleteComment` but I wanted to test both in one method

Reply:
Oh wasn't aware of that... I thought it is different... again learned something :)